winlogbeat/sys/event.go

```diff
@@ -35,7 +35,7 @@ type Event struct {
 	// System
 	Provider        Provider        `xml:"System>Provider"`
 	EventIdentifier EventIdentifier `xml:"System>EventID"`
-	Version         uint8           `xml:"System>Version"`
+	Version         float32         `xml:"System>Version"`
```

Comment:
I'm curious what produces a float here? Doesn't the schema dictate that this value is unsigned 8-bit integer?

https://docs.microsoft.com/en-us/windows/desktop/WES/schema-version-systempropertiestype-element

Reply:
Ive been using this to ingest Netapp logs, and they appear to use floats.

Reply:
Do you have a sample event in XML form where a float is used?

Reply:
Here is a sample log in xml format: https://send.firefox.com/download/4c3e3bd9c689e63d/#thga7PfYN-zZE1OwRWmbYg

Reply:
Does the Windows Event Viewer open the .evtx files produced by the NetApp software? Do you happen to have one of the .evtx files you can share too?  I think the NetApp software is not following the Microsoft schema.

I can't change the `Version` type without breaking backwards compatibility. This is because in the index template we have declared that the `version` field is a `long` so if Winlogbeat starts sending floating point data then Elasticsearch is going to reject it the event. But we should be able to make a change such that it can still unmarshal the XML, but ignore the value if it's not an unsigned 8-bit integer.

Reply:
Here is the evtx: https://send.firefox.com/download/d67dba52a710aca4/#RoYwmqqMygibhKns6Ru96g
Funnily enough, i generated that XML file from windows event viewer.

Reply:
Thanks. Then that's pretty good evidence that the our parser should be more lax.

Reply:
Would you prefer me to make a pr making things more lax, or would you like to do it?

Reply:
Here's the PR https://github.com/elastic/beats/pull/15839 to change the parser.